django_loci/static/django-loci/js/loci.js

```diff
@@ -377,4 +497,120 @@ django.jQuery(function ($) {
+            var currentMarkerCoords = getMarker() !== undefined ? getMarker().getLatLng() : null;
+            var currentMarkerLat = getMarker() !== undefined ? getMarker().getLatLng().lat : null;
+            var currentMarkerLng = getMarker() !== undefined ? getMarker().getLatLng().lng : null;
+            if (currentMarkerCoords !== null && window.markerCoords === null && !$addressInput.val()) {
+                window.markerCoords = currentMarkerCoords;
+                var reverseApiWithData = window.reverseApi + '?lat=' + currentMarkerLat + '&lng=' + currentMarkerLng;
+                window.fetch(reverseApiWithData)
+                    .then(function (resp) {
+                        var dataPromise = resp.json();
+                        dataPromise.then(function (data) {
+                            if (resp.status === 200) {
+                                $addressInput.val(data.address);
+                            } else {
+                                var message = 'We couldn\'t find a related address to the location '
+                                        + 'selected on the map, please enter the address manually';
+                                alert(message);
+                                window.skip_geocode = 2;
+                            }
+                        });
+                    });
+            }
+            if (getMarker() !== undefined) {
+                if (getMarker().dragging._enabled === true) {
+                    window.update = true;
+                    updateAdress();
+                } else {
+                    window.beforeEditLat = currentMarkerLat;
+                    window.beforeEditLng = currentMarkerLng;
+                }
+            }
+        }, 500);
+
+        var submit = false;
+        $form.submit(function (e) {
+            var currentMarkerCoords = getMarker() !== undefined ? getMarker().getLatLng() : null;
+            var currentMarkerLat = getMarker() !== undefined ? getMarker().getLatLng().lat : null;
+            var currentMarkerLng = getMarker() !== undefined ? getMarker().getLatLng().lng : null;
+            if (currentMarkerCoords !== window.markerCoords && currentMarkerCoords !== null &&
+                    getMarker().dragging._enabled !== true) {
+                var reverseApiWithData = window.reverseApi + '?lat=' + currentMarkerLat + '&lng=' + currentMarkerLng;
+                window.fetch(reverseApiWithData)
+                    .then(function (resp) {
+                        var dataPromise = resp.json();
+                        dataPromise.then(function (data) {
+                            if ($addressInput.val()) {
+                                if (currentMarkerLat !== window.addressLat ||
+                                        currentMarkerLng !== window.addressLng) {
+                                    if (resp.status === 200) {
+                                        if (window.confirm('Would you like to update address?')) {
```

Comment:
I think we should ask the user for confirmation before doing the HTTP request, otherwise it looks like kinda slow.
Best thing would be to do the request in parallel to asking for confirmation, that way it would result in very smooth UI.

Reply:
so behavior should be following:
Ask user for confirmation and while waiting for user's confirmation asynchronously do request to API, and then after affirmative answer to popup, instantly update address field?

Reply:
that would be awesome!

Reply:
Done :+1: